evennia/contrib/auditing/server.py

```diff
@@ -0,0 +1,266 @@
+                    data = kwargs['text'][0]
+                elif not 'text' in kwargs and len(kwargs.keys()) == 1:
+                    data = kwargs.keys()[0]
+                else:
+                    data = str(kwargs['text'])
+                    
+            except: data = str(kwargs)
+            
+        bytes = len(data.encode('utf-8'))
+                
+        data = data.strip()
+            
+        # Do not log empty lines
+        if not data: return {}
+        
+        # Get current session's IP address
+        client_ip = session.address
+        
+        # Capture Account name and dbref together
+        account = session.get_account()
+        account_token = ''
+        if account: 
+            account_token = '%s%s' % (account.key, account.dbref)
+            
+        # Capture Character name and dbref together
+        char = session.get_puppet()
+        char_token = ''
+        if char:
+            char_token = '%s%s' % (char.key, char.dbref)
+            
+        # Capture Room name and dbref together
+        room = None
+        room_token = ''
+        if char:
+            room = char.location
+            room_token = '%s%s' % (room.key, room.dbref)
+            
+        # Mask any PII in message, where possible
+        data = self.mask(data, **kwargs)
+            
+        # Compile the IP, Account, Character, Room, and the message.
+        log = {
+            'time': time_str,
+            'hostname': socket.getfqdn(),
+            'application': '%s' % ev_settings.SERVERNAME,
+            'version': get_evennia_version(),
+            'pid': os.getpid(),
+            'direction': 'SND' if src == 'server' else 'RCV',
+            'protocol': self.protocol_key,
+            'ip': client_ip,
+            'session': 'session#%s' % self.sessid,
+            'account': account_token,
+            'character': char_token,
+            'room': room_token,
+            'msg': '%s' % data,
+            'bytes': bytes,
+            'objects': {
+                'time': time_obj,
+                'session': self,
+                'account': account,
+                'character': char,
+                'room': room,
+            }
+        }
+
+        return log
+        
+    def mask(self, msg, **kwargs):
+        """
+        Masks potentially sensitive user information within messages before
+        writing to log. Recording cleartext password attempts is bad policy.
+    
+        Args:
+            msg (str): Raw text string sent from client <-> server
+    
+        Returns:
+            msg (str): Text string with sensitive information masked out.
+
+        """
+        # Get command based on fuzzy match
+        command = next((x for x in re.findall('^(?:Command\s\')*[\s]*([create]{5,6}|[connect]{6,7}|[@userpassword]{6,13}).*', msg, flags=re.IGNORECASE)), None)
+        if not command or command in AUDIT_MASK_IGNORE:
+            return msg
+            
+        # Break msg into terms
```

Comment:
As mentioned, I feel you can probably skip the manual handling of terms in favor of applying a regex for finding and masking the parts that should be removed.